src/rpc/budget.cpp

```diff
@@ -733,7 +733,9 @@ UniValue mnbudgetrawvote(const UniValue& params, bool fHelp)
     vote.SetVchSig(vchSig);
 
     if (!vote.CheckSignature(true)) {
-        return "Failure to verify signature.";
+        // try old message version
+        vote.nMessVersion = MessageVersion::MESS_VER_STRMESS;
+        if (!vote.CheckSignature(true)) return "Failure to verify signature.";
```

Comment:
Shouldn't this have the activation height validation to reject the message if the signature is not valid anymore?

```
if (!vote.CheckSignature(true))
    return "failure etc";
 if (Params().NewSigsActive(nChainHeight - 600))
    return "failure, old signatures not accepted anymore".
           
```

Reply:
the `nMessVersion` is set (height-based) during the signature.
In this RPC the signature is provided externally so the client doesn't know it. Old version shouldn't be rejected.